unlock-app/src/components/content/event/Form.tsx

```diff
@@ -0,0 +1,356 @@
+import { config } from '~/config/app'
+import { useState } from 'react'
+import { Lock, Token } from '@unlock-protocol/types'
+import { MetadataFormData } from '~/components/interface/locks/metadata/utils'
+import { FormProvider, useForm, Controller, useWatch } from 'react-hook-form'
+import {
+  Button,
+  Disclosure,
+  Input,
+  TextBox,
+  Select,
+  ToggleSwitch,
+} from '@unlock-protocol/ui'
+import { useConfig } from '~/utils/withConfig'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { networkDescription } from '~/components/interface/locks/Create/elements/CreateLockForm'
+import { useQuery } from '@tanstack/react-query'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+import { BalanceWarning } from '~/components/interface/locks/Create/elements/BalanceWarning'
+import { SelectCurrencyModal } from '~/components/interface/locks/Create/modals/SelectCurrencyModal'
+import { CryptoIcon } from '~/components/interface/locks/elements/KeyPrice'
+import { UNLIMITED_KEYS_DURATION } from '~/constants'
+
+export interface NewEventForm {
```

Comment:
@julien51 can use use zod here to align with other form also helps for validations eventually

Reply:
Great idea! Let me try to add/replace it!